### PR TITLE
[TS] LPS-174381 Date range is reset when clicking 'Refresh Counts'. If there is a range available in the request then prefer that one

### PR DESCRIPTION
**Pull request by jonathanmccann**
https://issues.liferay.com/browse/LPS-174381

This one was a little tricky so please let me know if you see a better way to resolve it.

The issue stems from the fact that clicking "Refresh Counts" reloads the page but we then still fetch information from the _exportImportConfiguration_. This causes a mismatch between what is being displayed in the UI and what is being sent back to the server. In these changes I check to see if the request object is populated, which only happens after a refresh, and if so then get the date range from the request instead of the old _exportImportConfiguration_.

Let me know if there are any questions or if you see a different way around this issue.

### CODE REVIEW

**Comment by moltam89** [comment]
Hi @jonathanmccann ,

I'm one step behind you, I opened https://issues.liferay.com/browse/LPS-174802, then realised that you already sent a fix for this, I'll check that PR as well.

I also opened https://issues.liferay.com/browse/LPS-174803 and I'd like to look into this before we merge this current fix.

I hope I can process everything by tomorrow and I'll get back to you.

Thanks,
Tamas

**Comment by moltam89** [comment]
Hi @jonathanmccann ,

**TLDR:
Both fixes are great, I'm just creating backlog for the Staging Team by opening Publish Templates related tickets.**
I'm forwarding the fixes to the QA team.

I quickly looked into [LPS-174803](https://issues.liferay.com/browse/LPS-174803), it looks like during page load we always fetch content as if "From Last Publish Date" was selected. Selecting a Template to edit, and creating a publication from the Template are affected.

About https://github.com/liferay-staging/liferay-portal/pull/232, I opened [LPS-174931](https://issues.liferay.com/browse/LPS-174931)  because I noticed that the calendar doesn't even let as select future dates, when we start a "Custom Publish Process". I tend to think this should be applied for Templates as well. (We can set only the time to one hour from now, so your fix is would be still necessary, even if the calendar did not let us select future dates )

I also opened [LPS-174932](https://issues.liferay.com/browse/LPS-174932) , because I think Templates should be more flexible and we should allow future dates.

Thank you for always submitting conscious fixes, I don't even remember the last time when I found something suspicious in your code, during these reviews I mostly here to learn from your solutions.

Tamas

**Comment by moltam89** [comment]
cc @gaborlovasdotcom 
I'm assigning the related ticket to you, for QA.

**Comment by gaborlovasdotcom** [comment]
ci:test:bundle

**Comment by gaborlovasdotcom** [comment]
ci:forward

**Comment by gaborlovasdotcom** [comment]
@JoyceWang08211 could you please check if the relevant failure is related?

**Comment by JoyceWang08211** [comment]
> @JoyceWang08211 could you please check if the relevant failure is related?

The relevant failure is not related to the current changes.
cc @gaborlovasdotcom

**Comment by JoyceWang08211** [comment]
ci:test:relevant

**Comment by JoyceWang08211** [comment]
It's safe to send this PR to BChan directly. The relevant failure is not related to the current changes.
cc @moltam89

**Comment by moltam89** [comment]
Sent to Brian directly:
https://github.com/brianchandotcom/liferay-portal/pull/129992
